src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -513,13 +513,7 @@ public static bool TryReadBlock(BinaryReader reader, bool saveExtraFieldsAndComm
             // of the ExtraField block. Thus we must force the stream's position to the proper place.
             reader.BaseStream.AdvanceToPosition(endExtraFields);
 
-            if (saveExtraFieldsAndComments)
-                header.FileComment = reader.ReadBytes(header.FileCommentLength);
-            else
-            {
-                reader.BaseStream.Position += header.FileCommentLength;
-                header.FileComment = null;
-            }
+            header.FileComment = reader.ReadBytes(header.FileCommentLength);
```

Comment:
Do we still need the parameter `saveExtraFieldsAndComments`?
Does this still holds true?
https://github.com/dotnet/runtime/blob/218c34788340b40e80342bdbfeda2c85cd76aa56/src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchive.cs#L533

Reply:
Yes, it's needed for other things. See lines [496-509](https://github.com/dotnet/runtime/blob/218c34788340b40e80342bdbfeda2c85cd76aa56/src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs#L496-L509).